pybwa/libbwaaln.pyx

```diff
@@ -356,15 +356,15 @@ cdef class BwaAln:
         rec.mapping_quality = seq.mapQ
 
         # cigar
-        cigar = ""
+        cigartuples = []
         if seq.cigar:
             for j in range(seq.n_cigar):
                 cigar_len = __cigar_len(seq.cigar[j])
-                cigar_op = "MIDS"[__cigar_op(seq.cigar[j])]
-                cigar = f"{cigar}{cigar_len}{cigar_op}"
+                cigar_op = __cigar_op(seq.cigar[j])
```

Comment:
`__cigar_op` comes from [bwa itself](https://github.com/lh3/bwa/blob/79b230de48c74156f9d3c26795a360fc5a2d5d3b/bwtaln.h#L55) with the values defined [here in bwa](https://github.com/lh3/bwa/blob/79b230de48c74156f9d3c26795a360fc5a2d5d3b/bwtaln.h#L31-L34)
```suggestion
                # NB: this relies on the fact that the operator values in bwa are the same as pysam. E.g.
                # See  `CIGAR_OPS` in `pysam.libcalignedsegment.pyx` and `FROM_[MIDS]` in
                # `bwtaln.h`
                cigar_op = __cigar_op(seq.cigar[j])
```